client/my-sites/domains/domain-management/edit/domain-types/site-redirect-type.jsx

```diff
@@ -0,0 +1,203 @@
+ */
+import React from 'react';
+import { connect } from 'react-redux';
+import { localize } from 'i18n-calypso';
+/**
+ * Internal dependencies
+ */
+import config from 'config';
+import { Card } from '@automattic/components';
+import { withLocalizedMoment } from 'components/localized-moment';
+import DomainStatus from '../card/domain-status';
+import SubscriptionSettings from '../card/subscription-settings';
+import { recordPaymentSettingsClick } from '../payment-settings-analytics';
+import {
+	getByPurchaseId,
+	hasLoadedSitePurchasesFromServer,
+	isFetchingSitePurchases,
+} from 'state/purchases/selectors';
+import NonPrimaryDomainPlanUpsell from '../../components/domain/non-primary-domain-plan-upsell';
+import RenewButton from 'my-sites/domains/domain-management/edit/card/renew-button';
+import AutoRenewToggle from 'me/purchases/manage-purchase/auto-renew-toggle';
+import QuerySitePurchases from 'components/data/query-site-purchases';
+import { shouldRenderExpiringCreditCard } from 'lib/purchases';
+import ExpiringCreditCard from '../card/notices/expiring-credit-card';
+import DomainManagementNavigation from '../navigation';
+import DomainManagementNavigationEnhanced from '../navigation/enhanced';
+import { WrapDomainStatusButtons } from './helpers';
+
+class SiteRedirectType extends React.Component {
+	resolveStatus() {
+		const { translate, purchase } = this.props;
+
+		if ( purchase && shouldRenderExpiringCreditCard( purchase ) ) {
+			return {
+				statusText: translate( 'Action required' ),
+				statusClass: 'status-error',
+				icon: 'info',
+			};
+		}
+
+		return {
+			statusText: translate( 'Active' ),
+			statusClass: 'status-success',
+			icon: 'check_circle',
+		};
+	}
+
+	renderDefaultRenewButton() {
+		const { domain, purchase } = this.props;
+
+		if ( ! domain.currentUserCanManage ) {
+			return null;
+		}
+
+		return (
+			<div>
+				<RenewButton
+					compact={ true }
+					purchase={ purchase }
+					selectedSite={ this.props.selectedSite }
+					subscriptionId={ parseInt( domain.subscriptionId, 10 ) }
+					tracksProps={ { source: 'site-redirect-url', domain_status: domain.name } }
+				/>
+			</div>
+		);
+	}
+
+	renderAutoRenewToggle() {
+		const { selectedSite, purchase } = this.props;
+
+		if ( ! purchase ) {
+			return null;
+		}
+
+		const content = (
+			<AutoRenewToggle
+				planName={ selectedSite.plan.product_name_short }
+				siteDomain={ selectedSite.domain }
+				purchase={ purchase }
+				compact={ true }
+				withTextStatus={ true }
+				toggleSource="site-redirect-status"
+			/>
+		);
+
+		return content && <WrapDomainStatusButtons>{ content }</WrapDomainStatusButtons>;
+	}
+
+	renderAutoRenew() {
+		const { isLoadingPurchase } = this.props;
+
+		if ( isLoadingPurchase ) {
+			return (
+				<WrapDomainStatusButtons className="domain-types__auto-renew-placeholder">
+					<p />
+				</WrapDomainStatusButtons>
+			);
+		}
+
+		return this.renderAutoRenewToggle();
+	}
+
+	planUpsellForNonPrimaryDomain() {
```

Comment:
I think we can remove this as site redirects should be outside the primary domains with plans only scope